### PR TITLE
[DO NOT MERGE] i18n improvements

### PR DESCRIPTION
**Pull request by DavidRouyer**
### Related issues

#1749 

### Short description and why it's useful

This is the current goal I want to achieve

- Put all translations in the theme folder
- Translations files are now JSON files (translation.preprocessor is not needed anymore)
- Use `i18n.setLocaleMessage('en-US', require('theme/resource/i18n/en-US.json'))` in the theme to bundle the default language and lazy load the others
- Remove unneeded config entries
- **TODO**: make the JSON file look more like the RFC one (I will simply create a `core` section to avoid breaking user's current translations, future work will be done in the v2 of the theme)

### Upgrade Notes and Changelog

- [ ] No upgrade steps required (100% backward compatibility)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature **TODO**

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

**Comment by pkarw** [comment]
Ok, I see this for v1.7.0 @filrak

**Comment by filrak** [comment]
@DavidRouyer do we really need to lazy load all languages? I mean we can even load them on demand (or on hover of language switcher, this is a nice way to do preloads 1-2 sec before click)

**Comment by pkarw** [comment]
We should preload only these langs that are set in the`config.storeViews` - it makes no sense to preload languages that user can not use anyway

File: core/i18n/index.ts [hunk 1]
```diff
@@ -19,27 +19,20 @@ function setI18nLanguage (lang: string): string {
 }
 
 export function loadLanguageAsync (lang: string): Promise<string> {
-  if (!config.i18n.bundleAllStoreviewLanguages) {
```
Comment (pkarw) [inline core/i18n/index.ts]: Wht this option has been removed? It was a solution for "lang blinking" and to not preload all the translations all the time

**Comment by DavidRouyer** [comment]
@filrak Then, how to load them on demand? 😄 That's the point of the `loadLanguageAsync()` which is called when you switch between stores.

I removed the line https://github.com/DivanteLtd/vue-storefront/blob/92310aa37d004ef8dac22f75af73354ce3739ec8/core/i18n/index.ts#L43 so they are not preloaded anymore.

@pkarw There is no "lang blinking" anymore since I `require()` the English language AND the defaultLocale language in the theme.

**Comment by pkarw** [comment]
@DavidRouyer it still can be a problem as user can use LanguageSwitcher to switch to any of the `storeViews` - this switch reloads the page: https://demo.storefrontcloud.io -> https://demo.storefrontcloud.io/de 

And then - when `de` is not defaultLocale - there will be blinking anyway. So we need to preload all the langs that are in the `config.storeViews.i18n` as it was done before. Please not remove this feature - it's necessary to keep it.

You Comment to @filrak is not apropriete because we're RELOADING the page when swithcing the lang (it's required to reload it by many reasons - clearing the cache/memory objects related to current locale is one of them)

**Pull request closed without merging**